internal/operator/backrest/repo.go

```diff
@@ -177,14 +184,14 @@ func CreateRepoSecret(clientset kubernetes.Interface, cluster *crv1.Pgcluster) e
 // as needed to support the creation of a bootstrap repository need to bootstrap a new cluster from an
 // existing data source.
 func setBootstrapRepoOverrides(clientset kubernetes.Interface, cluster *crv1.Pgcluster,
-	repoFields *RepoDeploymentTemplateFields) error {
+	repoFields *RepoDeploymentTemplateFields, namespace string) error {
 	ctx := context.TODO()
 
 	restoreClusterName := cluster.Spec.PGDataSource.RestoreFrom
-	namespace := cluster.GetNamespace()
 
 	repoFields.ClusterName = restoreClusterName
 	repoFields.BootstrapCluster = cluster.GetName()
+	repoFields.BootstrapNamespace = cluster.GetNamespace()
```

Comment:
This is a bit confusing. Can we get some comments about which namespace is which and what we're doing here?

Reply:
The function comment now explains the `namespace` parameter, and why it may differ from the namespace of the `cluster`.